### PR TITLE
Bring up Android and Browser cross-compilers on Windows

### PR DESCRIPTION
**Pull request by directhex**
Thus far, only Android x64 cross compiler - we have been very lax with separating our use of HOST_WINDOWS vs TARGET_WINDOWS, and attempting to cross compile for other architectures fails horribly. This should get us at least partial progress towards the goal of platform parity on Windows with Linux.

### CODE REVIEW

**Comment by directhex** [comment]
Ignore the number of commits, merging from master is always fun.

**Comment by directhex** [comment]
Failure appears to be because we use the 32-bit->64-bit cross compiler. On our 64-bit builds of 64-bit Mono for 64-bit Windows. SOMEWHERE IN THE UNIVERSE THIS IS THE RIGHT WAY TO DO COMPUTERS

File: src/mono/mono.proj [hunk 1]
```diff
@@ -396,6 +397,38 @@
       <MonoLibClang Condition="!$([MSBuild]::IsOSPlatform('OSX'))">$(EMSDK_PATH)/upstream/lib/libclang.so</MonoLibClang>
     </PropertyGroup>
 
+    <!-- Windows specific options -->
+    <ItemGroup Condition="'$(OS)' == 'Windows_NT'">
```
Comment (pranavkm) [inline src/mono/mono.proj:401]:
```suggestion
    <ItemGroup Condition="$([MSBuild]::IsOSPlatform('Windows'))">
```

**Comment by lewing** [comment]
Failure looks like https://github.com/dotnet/runtime/pull/48075

**Review by steveisok — APPROVED** [review]

**Comment by steveisok** [comment]
Looks good. Way to swim through the pathing issues 👍

**Pull request merged**